src/iarc7_motion/linear_motion_profile_generator.py

```diff
@@ -0,0 +1,209 @@
+        if self._start_motion_point is not None:
+            start_motion_point = self._start_motion_point
+            self._start_motion_point = None
+
+            # Check that the time is at least equal to the current time
+            # before generating a plan
+            if start_motion_point.header.stamp < rospy.Time.now():
+                start_motion_point.header.stamp = rospy.Time.now()
+            return start_motion_point
+
+        else:
+            # Make sure a starting point newer than the last sent time is sent
+            for i in range(1, len(self._last_motion_plan.motion_points)):
+                if self._last_motion_plan.motion_points[i].header.stamp > time:
+                    first_point = self._last_motion_plan.motion_points[i-1]
+                    second_point = self._last_motion_plan.motion_points[i]
+                    return interpolate_motion_points(first_point, second_point, time)
+
+            # A point was not sent before the buffer ran out
+            # Use the oldest and reset the timestamp
+            self._last_motion_plan.motion_points[-1].header.stamp = rospy.Time.now()
+            return self._last_motion_plan.motion_points[-1]
+
+    # Get a motion plan that attempts to achieve a given velocity target
+    def get_velocity_plan(self, velocity_command):
+
+        # Get the stating motion point for the time that the velocity is desired
+        start_point = self._get_start_point(velocity_command.target_twist.header.stamp)
+
+        self._last_stamp = velocity_command.target_twist.header.stamp
+
+        p_start = msg_to_np(start_point.motion_point.pose.position)
+        v_start = msg_to_np(start_point.motion_point.twist.linear)
+        v_desired = msg_to_np(velocity_command.target_twist.twist.linear)
+        v_delta = v_desired - v_start
+
+        # Assign a direction to the target acceleration
+        a_target = self._TARGET_ACCEL * v_delta / np.linalg.norm(v_delta)
+
+        # Find the time required to accelerate to the desired velocity
+        acceleration_time = min(np.linalg.norm(v_delta) / self._TARGET_ACCEL, self._PLAN_DURATION)
+
+        # Use the rest of the profile duration to hold the velocity
+        steady_velocity_time = self._PLAN_DURATION - acceleration_time
+
+        # Calculate the number of discrete steps spent in each state
+        accel_steps = np.floor(acceleration_time / self._PROFILE_TIMESTEP)
+        vel_steps = np.floor(steady_velocity_time / self._PROFILE_TIMESTEP)
+
+        # Initialize the velocities array with the starting velocity
+        velocities = [v_start]
+
+        # Generate velocities corresponding to the acceleration period
+        if accel_steps > 0:
+            accel_times = (np.linspace(0.0, accel_steps*self._PROFILE_TIMESTEP, accel_steps)
+                     * accel_steps
+                     * self._PROFILE_TIMESTEP)
+            velocities.extend([a_target * accel_time + v_start for accel_time in accel_times])
```

Comment:
This is going to add two timesteps at `v_start`

Reply:
Fixed